src/service/tempMailApi.js

```diff
@@ -6,11 +6,24 @@ class TemporalMailService {
   constructor() {
     this.BASE_API_URL = 'https://api.mail.tm';
     this.httpClient = require('axios');
+    this.emailDomain = '';
   }
+
+  async _getDomains() {
+    try {
+      const response = await this.httpClient.get(`${this.BASE_API_URL}/domains`);
+      const {domain} = response.data['hydra:member'];
+      this.emailDomain = domain;
+    } catch (err) {
+      throw new Error('Cant get domains for emails');
```

Comment:
Just throwing an unexpected Error?
You should create a custom exception in order to give better user feedback

Reply:
You should provide useful information, not just a closure fail